include/clap/helpers/plugin-proxy.hxx

```diff
@@ -234,6 +235,28 @@ namespace clap { namespace helpers {
       return _pluginGui->hide(&_plugin);
    }
 
+   /////////////////////////
+   // clap_plugin_latency //
+   /////////////////////////
+   template <MisbehaviourHandler h, CheckingLevel l>
+   bool PluginProxy<h, l>::canUseLatency() const noexcept {
+      if (!_pluginLatency)
+         return false;
+
+      if (_pluginLatency->get)
+          return true;
+
+      //pluginMisbehaving("clap_plugin_latency is partially implemented");
+      return false;
+   }
+
+   template <MisbehaviourHandler h, CheckingLevel l>
+   uint32_t PluginProxy<h, l>::latencyGet() const noexcept {
+      assert(canUseLatency());
+      ensureMainThread("latency.get");
+      return _pluginLatency->get(&_plugin);
```

Comment:
This needs to check that the plugin is in active state.

Reply:
Ah, this is not documented in `latency.h`. So we should add that in both places

Reply:
Yeah I'll add it to clap, but having a latency in sample while the sample rate isn't known to the plugin doesn't mean much ;-)

Reply:
I've added it.

Reply:
You can track the active state, we do it in plugin.hh, and this is the reason why someone wants to use the clap-helpers, for all the runtime checks.

Yes remove the const specifier.

If the plugin proxy has a reference to the host, then it can check for the main thread and audio thread.